### PR TITLE
Support keyword argument in classifier link

### PR DESCRIPTION
**Pull request by unnonouno**
To support keyword arguments in Classifier, I added `label_key` attribute to specify label variable. When a user specify `str` key, this class uses keyword argument as a label variable.

### CODE REVIEW

**Comment by niboshi** [comment]
Can you fix test failures?

**Comment by niboshi** [comment]
Doc test has failed. Probably because of `print` statements?

**Comment by unnonouno** [comment]
Oh sorry. I removed print.

File: tests/chainer_tests/links_tests/model_tests/test_classifier.py [hunk 1]
```diff
@@ -24,16 +24,18 @@ def __call__(self, y, t):
 @testing.parameterize(*testing.product({
     'accfun': [AccuracyWithIgnoreLabel(), None],
     'compute_accuracy': [True, False],
-    'x_num': [1, 2]
+    'x_num': [1, 2],
+    'label_key': [-1, 't'],
```
Comment (niboshi) [inline tests/chainer_tests/links_tests/model_tests/test_classifier.py]: Please also check
- positive value (esp. at the boundary)
- zero

File: chainer/links/model/classifier.py [hunk 2]
```diff
@@ -61,13 +74,25 @@ def __call__(self, *args):
...
+            del args[self.label_key]
+        elif isinstance(self.label_key, str):
+            if self.label_key not in kwargs:
+                msg = 'Label key "%s" is not found' % self.label_key
+                raise ValueError(msg)
+            t = kwargs[self.label_key]
+            del kwargs[self.label_key]
+
         self.y = None
         self.loss = None
```
Comment (unnonouno) [inline chainer/links/model/classifier.py]: Hum, it is not required, i think. I'll fix it another PR.

**Comment by unnonouno** [comment]
I fixed it.

**Comment by niboshi** [comment]
Please use `six.assertRaisesRegex`.

**Comment by unnonouno** [comment]
OK

**Comment by niboshi** [comment]
LGTM
